File: probfilter-core/src/main/java/com/c6h5no2/probfilter/crdt/CvRFilter.java
package com.c6h5no2.probfilter.crdt;

import com.c6h5no2.probfilter.pdsa.Filter;


/**
 * A convergent (i.e. state-based) replicated filter.
 */
public interface CvRFilter<E, T> extends Filter<E, T>, CvRDT<T> {
    default FluentCvRFilter<E> asFluent() {
        return new FluentCvRFilter<>(this);
    }
}

File: probfilter-core/src/main/scala/com/c6h5no2/probfilter/crdt/CvRFilter.scala
package com.c6h5no2.probfilter.crdt

import com.c6h5no2.probfilter.pdsa.Filter

/**
 * A convergent (i.e. state-based) replicated probabilistic filter.
 */
trait CvRFilter[E, T] extends Filter[E, T] with CvRDT[T] {
  @deprecated
  def asFluent(): FluentCvRFilter[E] = {
    new FluentCvRFilter[E](this)
  }
}
